### PR TITLE
Add a recipe for gst-plugins-base

### PR DESCRIPTION
**Pull request by ccordoba12**
Last dependency for Qt5 on Linux. Boy, this was fast!!

Another one for you @ocefpaf :-)

### CODE REVIEW

File: recipes/gst-plugins-base/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# gstreamer expects libffi to be in lib64 for some reason.
+mkdir $PREFIX/lib64
```
Comment (ocefpaf) [inline recipes/gst-plugins-base/build.sh]: This still feels hackish to me. Don't forget to open an issue in the feedstock so we can try to patch it later.

File: recipes/gst-plugins-base/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+build:
+  skip: true  # [win or osx]
+  number: 0
+
+requirements:
+  build:
+    - gstreamer 1.8*
+    - m4
+    - perl
+    - xz
```
Comment (ocefpaf) [inline recipes/gst-plugins-base/meta.yaml]: We are pinning this to `xz 5.2.*`

File: recipes/gst-plugins-base/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  url: https://gstreamer.freedesktop.org/src/gst-plugins-base/gst-plugins-base-{{ version }}.tar.xz
+  sha256: {{ sha256 }}
+
+build:
+  skip: true  # [win or osx]
+  number: 0
+
+requirements:
+  build:
+    - gstreamer 1.8*
```
Comment (ocefpaf) [inline recipes/gst-plugins-base/meta.yaml]: Maybe `gstreamer 1.8.*` is more appropriated?

**Comment by ccordoba12** [comment]
Three tests are failing. I'll have to check why :-)

**Comment by ccordoba12** [comment]
@jakirkham, I remember you mentioned a trick to print the contents of a file on failure of a command. Unfortunately I forgot it :-)

I could use it here to print the the output file generated by `make check` to see why it's failing :-)

**Comment by jakirkham** [comment]
Yep, it's probably time to add it to an example. 😄 See below.

File: recipes/gst-plugins-base/build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+cp -r $PREFIX/lib/libffi* $PREFIX/lib64
+
+# The datarootdir option places the docs into a temp folder that won't
+# be included in the package (it is about 12MB).
+./configure --disable-examples --prefix="$PREFIX" --datarootdir=`pwd`/tmpshare
+make
+make check
```
Comment (jakirkham) [inline recipes/gst-plugins-base/build.sh]: Please append `|| { cat tests/check/test-suite.log; exit 1;}`.

File: recipes/gst-plugins-base/build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+make install
+
+# Remove the created lib64 directory
+rm -rf $PREFIX/lib64
```
Comment (jakirkham) [inline recipes/gst-plugins-base/build.sh]: It doesn't install stuff in there does it?

File: recipes/gst-plugins-base/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    - test -f $PREFIX/lib/libgstriff-1.0.so
+    - test -f $PREFIX/lib/libgstrtp-1.0.so
+    - test -f $PREFIX/lib/libgstrtsp-1.0.so
+    - test -f $PREFIX/lib/libgstsdp-1.0.so
+    - test -f $PREFIX/lib/libgsttag-1.0.so
+    - test -f $PREFIX/lib/libgstvideo-1.0.so
+
+about:
+    home: http://gstreamer.freedesktop.org/
+    license: GPL2
```
Comment (jakirkham) [inline recipes/gst-plugins-base/meta.yaml]: License file? 😄

**Comment by ccordoba12** [comment]
I had to comment `make check` because tests don't pass if not all plugins are built. But we don't need all plugins for Qt5, just audio and video.

Please see this link for an explanation:

https://bugzilla.gnome.org/show_bug.cgi?id=752778#c17

**Comment by ccordoba12** [comment]
@ocefpaf, could you merge this one please?

**Comment by ocefpaf** [comment]
> @ocefpaf, could you merge this one please?

Done!

**Comment by ccordoba12** [comment]
Thanks man, always very reliable with that green button! ;-p

**Comment by ocefpaf** [comment]
I started reviewing before merging and than I had a [Deva Ju](https://github.com/conda-forge/staged-recipes/pull/1438#discussion_r76900047) and decided to just hit the button :smile:

File: recipes/gst-plugins-base/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/gst-plugins-base/meta.yaml]: Done below.

**Pull request merged**